sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -152,17 +152,17 @@ class AdaptiveQueryExecSuite extends QueryTest with SharedSparkSession {
       val (plan, adaptivePlan) = runAdaptiveAndVerifyResult(
         """
           |WITH t4 AS (
-          |  SELECT * FROM lowercaseData t2 JOIN testData3 t3 ON t2.n = t3.a
+          |  SELECT * FROM lowercaseData t2 JOIN testData3 t3 ON t2.n = t3.a where t2.n = '1'
```

Comment:
what does this change do? To provide one more chance to turn SMJ to BHJ?

Reply:
Yeah, without this change, SMJ wouldn't be changed to BHJ because of the new cost guard.